jcache/embedded/src/main/java/org/infinispan/jcache/embedded/marshall/SingletonExternalizer.java

```diff
@@ -0,0 +1,39 @@
+package org.infinispan.jcache.embedded.marshall;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Set;
+
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.commons.util.Util;
+
+public class SingletonExternalizer<T> implements AdvancedExternalizer<T> {
```

Comment:
Could this (and `SuppliedExternalizer`) be useful for other externalizers in other modules? If so we could have these class in core instead? WDYT?